docs/prometheus-integration.md

```diff
@@ -106,3 +106,143 @@ The configuration file above can be used to deploy Prometheus Server with
 scraping configuration for Antrea services.
 To deploy this configuration use
 `kubectl apply -f build/yamls/antrea-prometheus.yml`
+
+# Antrea Prometheus Metrics
+Antrea Controller and Agents expose various metrics, some are provided by the 
+Antrea components and others are provided by 3rd party components which are used
+by the Antrea components.
+
+Below is a list of metrics, provided by the components and by 3rd parties.
+
+## Antrea Controller Metrics
+**antrea_controller_address_group_processed:** The total number of address-group
+processed 
+**antrea_controller_address_group_sync_duration_milliseconds:** The duration of
+syncing address-group
+**antrea_controller_applied_to_group_processed:** The total number of
+applied-to-group processed
+**antrea_controller_applied_to_group_sync_duration_milliseconds:** The duration
+of syncing applied-to-group
+**antrea_controller_length_address_group_queue:** The length of
+AddressGroupQueue
+**antrea_controller_length_applied_to_group_queue:** The length of
+AppliedToGroupQueue
+**antrea_controller_length_network_policy_queue:** The length of
+InternalNetworkPolicyQueue
+**antrea_controller_network_policy_processed:** The total number of
+internal-networkpolicy processed
+**antrea_controller_network_policy_sync_duration_milliseconds:** The duration of
+syncing internal-networkpolicy
+**antrea_controller_runtime_info:** Antrea controller runtime info, defined as
+labels. The value of the gauge is always set to 1.
+
+## Antrea Agent Metrics
+**antrea_agent_egress_networkpolicy_rule_count:** Number of egress networkpolicy
+rules on local node which are managed by the Antrea Agent.
+**antrea_agent_ingress_networkpolicy_rule_count:** Number of ingress
+networkpolicy rules on local node which are managed by the Antrea Agent.
+**antrea_agent_local_pod_count:** Number of pods on local node which are managed
+by the Antrea Agent.
+**antrea_agent_networkpolicy_count:** Number of networkpolicies on local node
+which are managed by the Antrea Agent.
+**antrea_agent_ovs_flow_count:** Flow count for each OVS flow table. The TableID
+is used as a label.
+**antrea_agent_ovs_total_flow_count:** Total flow count of all OVS flow tables.
+**antrea_agent_runtime_info:** Antrea agent runtime info , defined as labels.
+The value of the gauge is always set to 1.
+
+## Common Metrics Provided by Infrastructure
+### Apiserver Metrics
```

Comment:
Was chatting with @antoninbas on this PR. He has a good point it is even harder to maintain 3rd party metrics here. If we can not point to any links for these metrics, we should consider auto-generating the list.
At least we can mention here, the list can change as the 3rd party components versions change.

Reply:
Hi @jianjuns, this list has been mostly autogenerated :)
I can complete it with a few things and commit the generation script into the repo (not sure where this belong though)

Reply:
`hack/`

There should also be a Github CI job to make sure that the list is up-to-date.